### ISSUE TITLE
Block style select props not updated on selection change 

### ISSUE THREAD

**Issue by alexeygolev** [sanity-io/sanity#2456]
**Describe the bug**

A regression introduced in 2.10.0 by [this line](https://github.com/sanity-io/sanity/blob/1bd6332831ce36692ba87a3ea8a9927ae24a9441/packages/%40sanity/form-builder/src/inputs/PortableText/Toolbar/Toolbar.tsx#L123). As far as I understand it has lost the `selection` dependency and I assume `editor` is not updated with selection thus not triggering a “fresh” function call

**To Reproduce**

1. Change a block style to H1 or H2
2. Put cursor in any other block
3. The "style" toolbar element doesn't change based on the cursor position

**Expected behavior**

The style in the block style select should reflect the style of the block that the cursor is currently in

**Screenshots**

**Which versions of Sanity are you using?**

```
@sanity/cli             2.10.0 (up to date)
@sanity/base            2.10.0 (up to date)
@sanity/components       2.2.6 (up to date)
@sanity/core            2.10.0 (up to date)
@sanity/default-layout  2.10.0 (up to date)
@sanity/default-login    2.8.0 (up to date)
@sanity/desk-tool       2.10.1 (up to date)
@sanity/vision          2.10.0 (up to date)

```

**Comment by fusepilot** [comment]
Seeing the same.

````
@sanity/base                2.10.2 (up to date)
@sanity/cli                 2.10.0 (up to date)
@sanity/components           2.2.6 (up to date)
@sanity/core                2.10.2 (up to date)
@sanity/default-layout      2.10.2 (up to date)
@sanity/default-login        2.8.0 (up to date)
@sanity/desk-tool           2.10.2 (up to date)
@sanity/production-preview   2.2.6 (up to date)
@sanity/vision              2.10.0 (up to date)
````

**Comment by geoffreyjball** [comment]
Resolved by `@sanity/desk-tool@2.10.4` (CC: @rexxars).

**Comment by rexxars** [comment]
As @geoffreyjball says, this was fixed in 2.10.4 - thanks for reporting!

**Comment by jasonbarone** [comment]
Thanks for the fix! Just wanted to drop a note for anyone finding this later on. In addition to the behavior mentioned above, this bug also prevented the editor from setting the default "Normal" style during content creation. Any text in your editor that was not explicitly set to a style will likely contain "No style" styles. If your serializer doesn't handle defaults, you may have unstyled text in your app. To fix, simply go into your documents and assign the style to Normal.

It looked like this:
 
<img width="245" alt="Screen Shot 2021-05-17 at 7 53 10 PM" src="https://user-images.githubusercontent.com/545055/118570924-105ea380-b74b-11eb-91d4-7ae0ac338c95.png">
